File: binding/src/libqgraphicstransform.cpp
#include <QGraphicsRotation>
#include <QGraphicsScale>
#include <QGraphicsTransform>
#include <QMatrix4x4>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QVector3D>
#include <qgraphicstransform.h>
#include "libqgraphicstransform.h"
#include "libqgraphicstransform.hxx"

QGraphicsTransform* QGraphicsTransform_new() {
	 return new VirtualQGraphicsTransform();
}

QGraphicsTransform* QGraphicsTransform_new2(QObject* parent) {
	 return new VirtualQGraphicsTransform(parent);
}

libqt_string QGraphicsTransform_Tr(const char* s) {
	QString _ret = QGraphicsTransform::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QGraphicsTransform_ApplyTo(const QGraphicsTransform* self, QMatrix4x4* matrix) {
	self->applyTo(matrix);
}

libqt_string QGraphicsTransform_Tr2(const char* s, const char* c) {
	QString _ret = QGraphicsTransform::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGraphicsTransform_Tr3(const char* s, const char* c, int n) {
	QString _ret = QGraphicsTransform::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
void QGraphicsTransform_QBaseApplyTo(const QGraphicsTransform* self, QMatrix4x4* matrix) {
	auto* vqgraphicstransform = dynamic_cast<const VirtualQGraphicsTransform*>(self);
	if (vqgraphicstransform && vqgraphicstransform->isVirtualQGraphicsTransform) {
vqgraphicstransform->setQGraphicsTransform_ApplyTo_IsBase(true);
	vqgraphicstransform->applyTo(matrix);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsTransform_OnApplyTo(const QGraphicsTransform* self, intptr_t slot) {
	auto* vqgraphicstransform = dynamic_cast<const VirtualQGraphicsTransform*>(self);
	if (vqgraphicstransform && vqgraphicstransform->isVirtualQGraphicsTransform) {
vqgraphicstransform->setQGraphicsTransform_ApplyTo_Callback(reinterpret_cast<VirtualQGraphicsTransform::QGraphicsTransform_ApplyTo_Callback>(slot));
}
}

void QGraphicsTransform_Delete(QGraphicsTransform* self) {
    delete self;
}

QGraphicsScale* QGraphicsScale_new() {
	 return new VirtualQGraphicsScale();
}

QGraphicsScale* QGraphicsScale_new2(QObject* parent) {
	 return new VirtualQGraphicsScale(parent);
}

libqt_string QGraphicsScale_Tr(const char* s) {
	QString _ret = QGraphicsScale::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QVector3D* QGraphicsScale_Origin(const QGraphicsScale* self) {
	return new QVector3D(self->origin());
}

void QGraphicsScale_SetOrigin(QGraphicsScale* self, const QVector3D* point) {
	self->setOrigin(*point);
}

double QGraphicsScale_XScale(const QGraphicsScale* self) {
	return self->xScale();
}

void QGraphicsScale_SetXScale(QGraphicsScale* self, double xScale) {
	self->setXScale(xScale);
}

double QGraphicsScale_YScale(const QGraphicsScale* self) {
	return self->yScale();
}

void QGraphicsScale_SetYScale(QGraphicsScale* self, double yScale) {
	self->setYScale(yScale);
}

double QGraphicsScale_ZScale(const QGraphicsScale* self) {
	return self->zScale();
}

void QGraphicsScale_SetZScale(QGraphicsScale* self, double zScale) {
	self->setZScale(zScale);
}

void QGraphicsScale_ApplyTo(const QGraphicsScale* self, QMatrix4x4* matrix) {
	self->applyTo(matrix);
}

void QGraphicsScale_OriginChanged(QGraphicsScale* self) {
	self->originChanged();
}

void QGraphicsScale_Connect_OriginChanged(QGraphicsScale* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsScale*) = reinterpret_cast<void (*)(QGraphicsScale*)>(slot);
    QGraphicsScale::connect(self, &QGraphicsScale::originChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsScale_XScaleChanged(QGraphicsScale* self) {
	self->xScaleChanged();
}

void QGraphicsScale_Connect_XScaleChanged(QGraphicsScale* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsScale*) = reinterpret_cast<void (*)(QGraphicsScale*)>(slot);
    QGraphicsScale::connect(self, &QGraphicsScale::xScaleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsScale_YScaleChanged(QGraphicsScale* self) {
	self->yScaleChanged();
}

void QGraphicsScale_Connect_YScaleChanged(QGraphicsScale* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsScale*) = reinterpret_cast<void (*)(QGraphicsScale*)>(slot);
    QGraphicsScale::connect(self, &QGraphicsScale::yScaleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsScale_ZScaleChanged(QGraphicsScale* self) {
	self->zScaleChanged();
}

void QGraphicsScale_Connect_ZScaleChanged(QGraphicsScale* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsScale*) = reinterpret_cast<void (*)(QGraphicsScale*)>(slot);
    QGraphicsScale::connect(self, &QGraphicsScale::zScaleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsScale_ScaleChanged(QGraphicsScale* self) {
	self->scaleChanged();
}

void QGraphicsScale_Connect_ScaleChanged(QGraphicsScale* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsScale*) = reinterpret_cast<void (*)(QGraphicsScale*)>(slot);
    QGraphicsScale::connect(self, &QGraphicsScale::scaleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QGraphicsScale_Tr2(const char* s, const char* c) {
	QString _ret = QGraphicsScale::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGraphicsScale_Tr3(const char* s, const char* c, int n) {
	QString _ret = QGraphicsScale::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
void QGraphicsScale_QBaseApplyTo(const QGraphicsScale* self, QMatrix4x4* matrix) {
	auto* vqgraphicsscale = dynamic_cast<const VirtualQGraphicsScale*>(self);
	if (vqgraphicsscale && vqgraphicsscale->isVirtualQGraphicsScale) {
vqgraphicsscale->setQGraphicsScale_ApplyTo_IsBase(true);
	vqgraphicsscale->applyTo(matrix);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsScale_OnApplyTo(const QGraphicsScale* self, intptr_t slot) {
	auto* vqgraphicsscale = dynamic_cast<const VirtualQGraphicsScale*>(self);
	if (vqgraphicsscale && vqgraphicsscale->isVirtualQGraphicsScale) {
vqgraphicsscale->setQGraphicsScale_ApplyTo_Callback(reinterpret_cast<VirtualQGraphicsScale::QGraphicsScale_ApplyTo_Callback>(slot));
}
}

void QGraphicsScale_Delete(QGraphicsScale* self) {
    delete self;
}

QGraphicsRotation* QGraphicsRotation_new() {
	 return new VirtualQGraphicsRotation();
}

QGraphicsRotation* QGraphicsRotation_new2(QObject* parent) {
	 return new VirtualQGraphicsRotation(parent);
}

libqt_string QGraphicsRotation_Tr(const char* s) {
	QString _ret = QGraphicsRotation::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QVector3D* QGraphicsRotation_Origin(const QGraphicsRotation* self) {
	return new QVector3D(self->origin());
}

void QGraphicsRotation_SetOrigin(QGraphicsRotation* self, const QVector3D* point) {
	self->setOrigin(*point);
}

double QGraphicsRotation_Angle(const QGraphicsRotation* self) {
	return self->angle();
}

void QGraphicsRotation_SetAngle(QGraphicsRotation* self, double angle) {
	self->setAngle(angle);
}

QVector3D* QGraphicsRotation_Axis(const QGraphicsRotation* self) {
	return new QVector3D(self->axis());
}

void QGraphicsRotation_SetAxis(QGraphicsRotation* self, const QVector3D* axis) {
	self->setAxis(*axis);
}

void QGraphicsRotation_SetAxis2(QGraphicsRotation* self, int axis) {
	self->setAxis(static_cast<Qt::Axis>(axis));
}

void QGraphicsRotation_ApplyTo(const QGraphicsRotation* self, QMatrix4x4* matrix) {
	self->applyTo(matrix);
}

void QGraphicsRotation_OriginChanged(QGraphicsRotation* self) {
	self->originChanged();
}

void QGraphicsRotation_Connect_OriginChanged(QGraphicsRotation* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsRotation*) = reinterpret_cast<void (*)(QGraphicsRotation*)>(slot);
    QGraphicsRotation::connect(self, &QGraphicsRotation::originChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsRotation_AngleChanged(QGraphicsRotation* self) {
	self->angleChanged();
}

void QGraphicsRotation_Connect_AngleChanged(QGraphicsRotation* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsRotation*) = reinterpret_cast<void (*)(QGraphicsRotation*)>(slot);
    QGraphicsRotation::connect(self, &QGraphicsRotation::angleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QGraphicsRotation_AxisChanged(QGraphicsRotation* self) {
	self->axisChanged();
}

void QGraphicsRotation_Connect_AxisChanged(QGraphicsRotation* self, intptr_t slot) {
    void (*slotFunc)(QGraphicsRotation*) = reinterpret_cast<void (*)(QGraphicsRotation*)>(slot);
    QGraphicsRotation::connect(self, &QGraphicsRotation::axisChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QGraphicsRotation_Tr2(const char* s, const char* c) {
	QString _ret = QGraphicsRotation::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QGraphicsRotation_Tr3(const char* s, const char* c, int n) {
	QString _ret = QGraphicsRotation::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
void QGraphicsRotation_QBaseApplyTo(const QGraphicsRotation* self, QMatrix4x4* matrix) {
	auto* vqgraphicsrotation = dynamic_cast<const VirtualQGraphicsRotation*>(self);
	if (vqgraphicsrotation && vqgraphicsrotation->isVirtualQGraphicsRotation) {
vqgraphicsrotation->setQGraphicsRotation_ApplyTo_IsBase(true);
	vqgraphicsrotation->applyTo(matrix);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsRotation_OnApplyTo(const QGraphicsRotation* self, intptr_t slot) {
	auto* vqgraphicsrotation = dynamic_cast<const VirtualQGraphicsRotation*>(self);
	if (vqgraphicsrotation && vqgraphicsrotation->isVirtualQGraphicsRotation) {
vqgraphicsrotation->setQGraphicsRotation_ApplyTo_Callback(reinterpret_cast<VirtualQGraphicsRotation::QGraphicsRotation_ApplyTo_Callback>(slot));
}
}

void QGraphicsRotation_Delete(QGraphicsRotation* self) {
    delete self;
}

File: binding/include/libqgraphicstransform.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgraphicstransform.html
pub const qgraphicstransform = struct {

    /// New constructs a new QGraphicsTransform object.
    pub fn New() QtC.QGraphicsTransform {
        return qtc.QGraphicsTransform_new();
    }


    /// New2 constructs a new QGraphicsTransform object.
    pub fn New2(parent: ?*anyopaque) QtC.QGraphicsTransform {
        return qtc.QGraphicsTransform_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsTransform_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicstransform.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn ApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsTransform_ApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    /// Allows for overriding the related default method
    pub fn OnApplyTo(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsTransform_OnApplyTo(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsTransform_QBaseApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    pub fn Update(self: ?*anyopaque, ) void {
        qtc.QGraphicsTransform_Update(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdate(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsTransform_OnUpdate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdate(self: ?*anyopaque, ) void {
        qtc.QGraphicsTransform_QBaseUpdate(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsTransform_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicstransform.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsTransform_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicstransform.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsTransform_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qgraphicsscale.html
pub const qgraphicsscale = struct {

    /// New constructs a new QGraphicsScale object.
    pub fn New() QtC.QGraphicsScale {
        return qtc.QGraphicsScale_new();
    }


    /// New2 constructs a new QGraphicsScale object.
    pub fn New2(parent: ?*anyopaque) QtC.QGraphicsScale {
        return qtc.QGraphicsScale_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsScale_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsscale.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Origin(self: ?*anyopaque, ) QtC.QVector3D {
        return qtc.QGraphicsScale_Origin(@ptrCast(self));
    }

    pub fn SetOrigin(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QGraphicsScale_SetOrigin(@ptrCast(self), @ptrCast(point));
    }

    pub fn XScale(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsScale_XScale(@ptrCast(self));
    }

    pub fn SetXScale(self: ?*anyopaque, xScale: f64) void {
        qtc.QGraphicsScale_SetXScale(@ptrCast(self), xScale);
    }

    pub fn YScale(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsScale_YScale(@ptrCast(self));
    }

    pub fn SetYScale(self: ?*anyopaque, yScale: f64) void {
        qtc.QGraphicsScale_SetYScale(@ptrCast(self), yScale);
    }

    pub fn ZScale(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsScale_ZScale(@ptrCast(self));
    }

    pub fn SetZScale(self: ?*anyopaque, zScale: f64) void {
        qtc.QGraphicsScale_SetZScale(@ptrCast(self), zScale);
    }

    pub fn ApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsScale_ApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    /// Allows for overriding the related default method
    pub fn OnApplyTo(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_OnApplyTo(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsScale_QBaseApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    pub fn OriginChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsScale_OriginChanged(@ptrCast(self));
    }

    pub fn OnOriginChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_Connect_OriginChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn XScaleChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsScale_XScaleChanged(@ptrCast(self));
    }

    pub fn OnXScaleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_Connect_XScaleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn YScaleChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsScale_YScaleChanged(@ptrCast(self));
    }

    pub fn OnYScaleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_Connect_YScaleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ZScaleChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsScale_ZScaleChanged(@ptrCast(self));
    }

    pub fn OnZScaleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_Connect_ZScaleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ScaleChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsScale_ScaleChanged(@ptrCast(self));
    }

    pub fn OnScaleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsScale_Connect_ScaleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsScale_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsscale.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsScale_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsscale.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsScale_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qgraphicsrotation.html
pub const qgraphicsrotation = struct {

    /// New constructs a new QGraphicsRotation object.
    pub fn New() QtC.QGraphicsRotation {
        return qtc.QGraphicsRotation_new();
    }


    /// New2 constructs a new QGraphicsRotation object.
    pub fn New2(parent: ?*anyopaque) QtC.QGraphicsRotation {
        return qtc.QGraphicsRotation_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsRotation_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsrotation.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Origin(self: ?*anyopaque, ) QtC.QVector3D {
        return qtc.QGraphicsRotation_Origin(@ptrCast(self));
    }

    pub fn SetOrigin(self: ?*anyopaque, point: ?*anyopaque) void {
        qtc.QGraphicsRotation_SetOrigin(@ptrCast(self), @ptrCast(point));
    }

    pub fn Angle(self: ?*anyopaque, ) f64 {
        return qtc.QGraphicsRotation_Angle(@ptrCast(self));
    }

    pub fn SetAngle(self: ?*anyopaque, angle: f64) void {
        qtc.QGraphicsRotation_SetAngle(@ptrCast(self), angle);
    }

    pub fn Axis(self: ?*anyopaque, ) QtC.QVector3D {
        return qtc.QGraphicsRotation_Axis(@ptrCast(self));
    }

    pub fn SetAxis(self: ?*anyopaque, axis: ?*anyopaque) void {
        qtc.QGraphicsRotation_SetAxis(@ptrCast(self), @ptrCast(axis));
    }

    pub fn SetAxis2(self: ?*anyopaque, axis: i32) void {
        qtc.QGraphicsRotation_SetAxis2(@ptrCast(self), @intCast(axis));
    }

    pub fn ApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsRotation_ApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    /// Allows for overriding the related default method
    pub fn OnApplyTo(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsRotation_OnApplyTo(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseApplyTo(self: ?*anyopaque, matrix: ?*anyopaque) void {
        qtc.QGraphicsRotation_QBaseApplyTo(@ptrCast(self), @ptrCast(matrix));
    }

    pub fn OriginChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsRotation_OriginChanged(@ptrCast(self));
    }

    pub fn OnOriginChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsRotation_Connect_OriginChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AngleChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsRotation_AngleChanged(@ptrCast(self));
    }

    pub fn OnAngleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsRotation_Connect_AngleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AxisChanged(self: ?*anyopaque, ) void {
        qtc.QGraphicsRotation_AxisChanged(@ptrCast(self));
    }

    pub fn OnAxisChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsRotation_Connect_AxisChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsRotation_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsrotation.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QGraphicsRotation_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qgraphicsrotation.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsRotation_Delete(@ptrCast(self));
    }
};
